File: backend/src/main/java/com/project2/domain/post/mapper/ToggleMapper.java
package com.project2.domain.post.mapper;

import org.springframework.stereotype.Component;

import com.project2.domain.member.entity.Member;
import com.project2.domain.post.entity.Likes;
import com.project2.domain.post.entity.Post;
import com.project2.domain.post.entity.Scrap;

@Component
public class ToggleMapper {

    public Likes toLikes(Long userId, Long postId) {
		return Likes.builder()
			.post(Post.builder().id(postId).build())
			.member(Member.builder().id(userId).build())
			.build();
    }

    public Scrap toScrap(Long userId, Post post) {
        return Scrap.builder()
                .post(post)
                .member(Member.builder().id(userId).build())
                .build();
    }
}

File: backend/src/main/kotlin/com/project2/domain/post/mapper/ToggleMapper.kt
package com.project2.domain.post.mapper

import com.project2.domain.member.entity.Member
import com.project2.domain.post.entity.Likes
import com.project2.domain.post.entity.Post
import com.project2.domain.post.entity.Scrap
import org.springframework.stereotype.Component

@Component
class ToggleMapper {

    fun toLikes(userId: Long, postId: Long): Likes {
        val post = Post().apply { id = postId }
        val member = Member().apply { id = userId }
        return Likes().apply {
            this.post = post
            this.member = member
        }
    }

    fun toScrap(userId: Long, post: Post): Scrap {
        val member = Member().apply { id = userId }
        return Scrap(post = post, member = member)
    }
}